### ISSUE TITLE
Intellj junit 5 runner does not show failed Steps with assertions.

### ISSUE THREAD

**Issue by abhishek-rdm** [serenity-bdd/serenity-core#2920]
Hi I am trying to run my serenity tests using Intellj (right click and run tests)

However, I have observed that if my test has a display name and the **assertion is used in a step**, IntelliJ somehow ignores the failed assertion.

<img width="1331" alt="Screenshot 2022-09-13 at 15 51 47" src="https://user-images.githubusercontent.com/66309037/189919435-dbdc2629-8ad1-4801-8fa6-53cbfb3a54d7.png">

```
@ExtendWith(SerenityJUnit5Extension.class)
class WhenSearchingForTerms {

    /**
     * Define the webdriver instance to be used for these tests
     */
    @Managed(driver = "chrome", options = "headless")
    WebDriver driver;

    /**
     * Navigation actions. This is a UIInteraction class so it will be instantiated automatically by Serenity.
     */
    NavigateActions navigate;

    /**
     * Actions related to searches. This is a UIInteraction class so it will be instantiated automatically by Serenity.
     */
    SearchActions search;

    /**
     * A page object representing a Wikipedia article that is currently appearing in the browser.
     * Page Objects are automatically initialised by Serenity.
     */
    DisplayedArticle displayedArticle;

    @Test
    @DisplayName("Test Display name")
    void searchBySingleKeyword() {
        search.testAssertion();
    }
}
```
<img width="934" alt="Screenshot 2022-09-13 at 15 54 28" src="https://user-images.githubusercontent.com/66309037/189919925-217f3fcc-9639-4c7d-a452-7baa2c41af3c.png">

Could you please help?

**Comment by wakaleo** [comment]
That's odd. @cliviu does this ring any bells? IntelliJ and Maven are not picking up the failures, though they appear correctly in the reports.

**Comment by cliviu** [comment]
Hey @abhishek-rdm , I can reproduce your problem. Until we fix it, as a workaround try to avoid setting the @DisplayName or add a static initializer in your test class , somethiing like   
   ```
@BeforeAll
    public static void initialize(){
        Serenity.throwExceptionsImmediately();
    }
```

**Comment by abhishek-rdm** [comment]
@cliviu thanks the workaround also works for me.

**Comment by bachhavdipak** [comment]
I have verified this issue and its resolved as a part of #3158 PR. I am waiting for author to go through code and accept my PR.

![image](https://github.com/serenity-bdd/serenity-core/assets/77110618/3ca507ae-2b7c-4886-9e70-35268160a9d0)
